Components/Sources/ComponentsObjC/WKSourceEditorFormatterStrikethrough.m

```diff
@@ -98,7 +98,7 @@ - (BOOL)attributedString:(NSMutableAttributedString *)attributedString isStriket
                isContentKey = YES;
            } else {
                // Edge case, check previous character if we are up against closing string
-               if (attrs[WKSourceEditorCustomKeyColorGreen]) {
+               if (attrs[WKSourceEditorCustomKeyColorGreen] && range.location < 0) {
```

Comment:
The crash came back when adding `attributedString.length > range.location - 1`, I recorded a video[1] to make it easier to understand when the app crashes. I'm not sure I'm missing something, but I'll revert this commit so I don't reintroduce the original bug and try to find another solution.

[1] https://drive.google.com/file/d/1dEYmx4z7ZvlSJem4wSLf0-gudLRTDngi/view?usp=sharing

Reply:
@mazevedofs That crash is in the reference formatter, looking at the call stack in your video. I think it probably just needs the same safety check of `&& attributedString.length > range.location - 1` at [this line](https://github.com/wikimedia/wikipedia-ios/blob/main/Components/Sources/ComponentsObjC/WKSourceEditorFormatterReference.m#L149). Every formatter will be given a chance to evaluate whether it should spit out a selected state whenever the cursor moves, so that's why that reference code is run here.

Reply:
That's it, then! I was testing each formatter individually, assuming it would only verify the string if it were in use. I'm investigating the test failures, but other than that, I think it's ready for another round of reviews.